java/ql/lib/semmle/code/java/security/CleartextStorageAndroidFilesystemQuery.qll

```diff
@@ -20,7 +20,7 @@ private class AndroidFilesystemCleartextStorageSink extends CleartextStorageSink
 /** A call to a method or constructor that may write to files to the local filesystem. */
 class LocalFileOpenCall extends Storable {
   LocalFileOpenCall() {
-    this = any(DataFlow::Node sink | sinkNode(sink, "create-file")).asExpr().(Argument).getCall()
+    this = any(DataFlow::Node sink | sinkNode(sink, "path-injection")).asExpr().(Argument).getCall()
```

Comment:
This seems that will change behavior, doesn't it? Now this will include the old `read-file` sinks.

Reply:
Hmm, yes, do you think we need to have a separate kind like "create-file" after all? (I suppose this is what you were referring to above when you said "I'm not sure we can find an easy solution for them.")

Reply:
To not go back to action-sounding kinds + difficult to map to a query kinds, what I can think of is turning this kind into an "advanced" one by doing something like `path-injection-create`/`path-injection-read` or `path-injection[create]`/`path-injection[read]`. In a way that queries uninterested in the specifics can just use `path-injection%`, but at the same time we keep the granularity for queries that need it.

Also this makes me think whether we should have some kind of native CodeQL support for this, something like:

```ql
predicate sinkNode(DataFlow::Node sink, string kind, string subkind)
```

Of course, that last part shouldn't be handled in this PR, just thinking out loud.

Reply:
Summarizing discussion from elsewhere: I'll run DCA/MRVA next week to see if adding `read-file` sinks causes issues with these queries.